packages/@aws-cdk/aws-ecs/lib/base/from-service-attributes.ts

```diff
@@ -32,22 +37,38 @@ export function fromServiceAtrributes(scope: Construct, id: string, attrs: Servi
     throw new Error('You can only specify either serviceArn or serviceName.');
   }
 
+  const featureConstruct = new CoreConstruct(scope, randomString());
+  const newArnFormat = FeatureFlags.of(featureConstruct).isEnabled(ECS_ARN_FORMAT_INCLUDES_CLUSTER_NAME);
```

Comment:
I tried `FeatureFlags.of(scope)` first because it should be the preferred solution. Unfortunately, it causes this error:

```
Argument of type 'import("/Users/julian/dev/aws-cdk/node_modules/constructs/lib/construct").Construct' is not assignable to parameter of type 'import("/Users/julian/dev/aws-cdk/packages/@aws-cdk/core/lib/construct-compat").Construct'.
Type 'Construct' is missing the following properties from type 'Construct': node, validate, prepare, synthesizets(2345)
```

Reply:
Since FeatureFlags expects a `Construct` from the core library, I think we can just switch to using that one in the `fromServiceAttributes()` function. I pushed this change to your branch, waiting for the build to see if this has other impacts.